### ISSUE TITLE
Hampton Bay Ceiling fan and light entities get same unique ID

### ISSUE THREAD

**Issue by twieder** [jason0x43/hacs-hubitat#46]
In versions 0.5.x, the Hampton Bay Ceiling Fan creates two entities in one device, but both get the same Unique_Id value. Because of this, the fan entity is unavailable -- everything pointing to fan.* returns the light.* entity.

**Comment by jason0x43** [comment]
Ok, I see what's happening there.

Could you check what capabilities your device is exposing? From a mac/linux command line, you can do

```
$ curl 'http://HUBITAT_ADDRESS/apps/api/MAKER_API_ID/devices/FAN_ID?access_token=TOKEN&prettyPrint=true'
```

There should be a `"capabilities"` property in the output with a list of things like "SwitchLevel". I'm curious as to what all is in the capabilities list.

**Comment by twieder** [comment]
Good morning,
I appreciate how responsive you are! 
Here is the response from the Hubitat MakerAPI:
[
    {
        "capabilities": [
            "Switch",
            {
                "attributes": [
                    {
                        "name": "switch",
                        "currentValue": "off",
                        "dataType": "ENUM",
                        "values": [
                            "on",
                            "off"
                        ]
                    }
                ]
            },
            "Configuration",
            "SwitchLevel",
            {
                "attributes": [
                    {
                        "name": "level",
                        "currentValue": "100",
                        "dataType": "NUMBER",
                        "values": null
                    }
                ]
            },
            "Refresh",
            "Actuator",
            "Light",
            {
                "attributes": [
                    {
                        "name": "switch",
                        "currentValue": "off",
                        "dataType": "ENUM",
                        "values": [
                            "on",
                            "off"
                        ]
                    }
                ]
            },
            "FanControl",
            {
                "attributes": [
                    {
                        "name": "speed",
                        "currentValue": "off",
                        "dataType": "ENUM",
                        "values": [
                            "low",
                            "medium-low",
                            "medium",
                            "medium-high",
                            "high",
                            "on",
                            "off",
                            "auto"
                        ]
                    }
                ]
            }
        ]
    }
]

Thanks,
Todd Wieder
4608 Quincy Adams Ct • Lorain, Ohio • 44053440-220-5680
toddwieder@yahoo.com
http://www.linkedin.com/in/todd-wieder 

    On Tuesday, May 19, 2020, 08:20:02 AM EDT, Jason Cheatham <notifications@github.com> wrote:  
 
 


Ok, I see what's happening there.

Could you check what capabilities your device is exposing? From a mac/linux command line, you can do
$ curl 'http://HUBITAT_ADDRESS/apps/api/MAKER_API_ID/devices/FAN_ID?access_token=TOKEN&prettyPrint=true'

There should be a "capabilities" property in the output with a list of things like "SwitchLevel". I'm curious as to what all is in the capabilities list.

**Comment by jason0x43** [comment]
Interesting. Does the fan provide both a switchable light and a fan control? Is the light dimmable, or just on/off?

**Comment by twieder** [comment]
The device exposes both a dimmable light and a fan. 
![Annotation 2020-05-19 100606](https://user-images.githubusercontent.com/14042013/82336311-6671fd00-99b8-11ea-9067-4811b003e008.png)

**Comment by jason0x43** [comment]
Ok, I had assumed that fans would show up as SwitchLevel devices (like dimmers), and that lowering the level would affect the fan speed. After seeing a couple fans, though, it seems that fans have discrete levels, and should always have a Fan capability.

So for something that has Fan, Light, and SwitchLevel, SwitchLevel applies to the light, not the fan.

Also, single devices can be both lights and fans, which the integration doesn't account for (but it really should).

That 0.5.2 release I just did was a bit premature. 🤦

I'll work on a fix for this. The integration will need to create unique entity IDs for for lights and fans, and it will need some more entity ID migration code to move existing IDs to the new format.

**Comment by jason0x43** [comment]
This was resolved in 32939b7 and included in v0.5.4.
